### DISCUSSION TITLE
Relations not being populated have the ID in their value

### DISCUSSION THREAD

**Discussion by Luxiorawa** [mikro-orm/mikro-orm#3518] [Q&A] [answered]
**Describe the bug**

**Note that i'm using Nest.js, but i guess it works the same as in the default library.**

When you retrieve one or multiple entities which have either a `ManyToOne` or a `OneToOne` relations but you don't populate this relation (which means, you either don't need it, or you don't want to return it), it keeps the field, but with the ID in it's value.

**To Reproduce**

I successfully reproduced the issue with the example repo about nest.js here : 

https://github.com/mikro-orm/nestjs-realworld-example-app

There is `Author` and `Articles`. An article can have on Author, but Author can have many articles. So the relations from Articles to Author is ManyToOne.

As defined in the Articles entity : 

article.entity.ts line 43
```typescript
  @ManyToOne()
  author: User;
```

If you call findOne method in article.controller.ts : **/GET /articles/{slug}** you'll end up calling findOne method in articles.services.ts which have the following definition : 

article.controller.ts line 103
```typescript
async findOne(userId: number, where): Promise<IArticleRO> {
    const user = userId ? await this.userRepository.findOneOrFail(userId, { populate: ['followers', 'favorites'] }) : undefined;
    const article = await this.articleRepository.findOne(where, { populate: ['author'] });
    return { article: article && article.toJSON(user) };
}
```

By default it return the article with it's corresponding author. So the response is something like this : 

```json
{
    "article": {
        "id": 1,
        "slug": "azeza",
        "title": "rqzedzqd",
        "description": "cwscswcws",
        "body": "sqdsq",
        "createdAt": "2022-05-29T22:00:00.000Z",
        "updatedAt": "2022-04-30T22:00:00.000Z",
        "tagList": [
            "test"
        ],
        "favoritesCount": 5,
        "author": {
            "id": 1,
            "username": "qzdqzdz",
            "bio": "1",
            "image": " ",
            "following": false
        },
        "comments": [],
        "favorited": false
    }
}
```

But if you delete the `{ populate: ['author'] }` and change few things so that it can be launched like so : 

```typescript
  async findOne(userId: number, where): Promise<any> {
    const article = await this.articleRepository.findOne(where);
    return { article: article };
  }
```

And comment out the content of the toJSON(user?: User) function like so : 

**article.entity.ts line 66 :**
```typescript
  toJSON(user?: User) {
    const o = wrap<Article>(this).toObject() as ArticleDTO;
    // o.favorited =
    //   user && user.favorites.isInitialized()
    //     ? user.favorites.contains(this)
    //     : false;
    // o.author = this.author.toJSON(user);

    return o;
  }
```

The response will be the following : 

```json
{
    "article": {
        "id": 1,
        "slug": "azeza",
        "title": "rqzedzqd",
        "description": "cwscswcws",
        "body": "sqdsq",
        "createdAt": "2022-05-29T22:00:00.000Z",
        "updatedAt": "2022-04-30T22:00:00.000Z",
        "tagList": [
            "test"
        ],
        "favoritesCount": 5,
        "author": 1,
        "comments": []
    }
}
```

In the author props, we have the ID of the author. 

**Expected behavior**

The props "author" either shouldn't be there, or it could be replaced by the raw database data, like author_id: 1 or something like that. 

It make more sense to me to either don't have the props if we don't want it, or either only have the "id" informative part.

**Versions**

| Dependency | Version |
| - | - |
| node | 16.13.0 |
| typescript | 4.7.4 |
| mikro-orm | 5.4.2 |
| nest.js | 9.0.11

**Comment by B4nan** [comment, marked as answer]
This works as intended, the value is selected, so you have it in your response. The value should be a naked FK value if not populated. You can control the `serializedName` if you want, by default it is the same as the property name. You can mark it as `hidden` or implement custom serializer - or alternatively mark it asi `lazy` so it gets omitted from the select clause unless you populate it.

Or if your issue is about something else, please try to be more specific, you started explaining how a complex realworld example app works, for reproductions you need to be as simple as possible, opposite of the realworld example app.

**Reply by Luxiorawa** [reply]
Your reply is on point, my issue was exactly that. Thanks for the response and the explanation !
